lib/fcm/message.dart

```diff
@@ -0,0 +1,585 @@
+import 'package:meta/meta.dart';
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'message.g.dart';
+
+/// https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages
+@JsonSerializable(explicitToJson: true, includeIfNull: false)
+class V1Message {
```

Comment:
It's kind of silly to name them V1Whatevers. What you can do here is name them something like FCMMessage, FCMError, etc. If the names do NOT conflict with ANYTHING else, you can just use Message.

Just please do away with the V1 thing everywhere

Reply:
Fcm has several protocols: v1 HTTP (recommended and implemented here), XMPP (deprecated) and legacy HTTP server (deprecated).
I'm adding a comment instead to precise it's using the HTTP v1 protocol. Other libraries I found use the deprecated legacy HTTP server protocol.